searchlib/src/vespa/searchlib/queryeval/nearest_neighbor_blueprint.cpp

```diff
@@ -84,10 +87,15 @@ NearestNeighborBlueprint::set_global_filter(const GlobalFilter &global_filter)
 {
     _global_filter = global_filter.shared_from_this();
     auto nns_index = _attr_tensor.nearest_neighbor_index();
+    LOG(debug, "set_global_filter with: %s / %s / %s",
+        (_approximate ? "approximate" : "exact"),
+        (nns_index ? "nns_index" : "no_index"),
+        (_global_filter->has_filter() ? "has_filter" : "no_filter"));
     if (_approximate && nns_index) {
         uint32_t est_hits = _attr_tensor.getNumDocs();
         if (_global_filter->has_filter()) {
             uint32_t max_hits = _global_filter->filter()->countTrueBits();
+            LOG(debug, "set_global_filter getNumDocs: %u / max_hits %u", est_hits, max_hits);
             if (max_hits * 10 < est_hits) {
```

Comment:
Until we have data on performance tests, perhaps we should _not_ do fallback to brute force. That will make it harder to interpret the results.